### ISSUE TITLE
Unexpected behavior when using ParticlePlot

### ISSUE THREAD

**Issue by cphyc** [yt-project/yt#1680]
### Bug report

**Bug summary**

When doing a ParticlePlot with positions, one expects that the outcome of it is a `ParticleProjectionPlot` object. However, depending on the order of the arguments, the user either get a `ParticleProjectionPlot`  or a `ParticlePhasePlot`.

**Code for reproduction**
For example using http://yt-project.org/data/DICEGalaxyDisk.tar.gz (but this is not specific to RAMSES)
```python
ds = yt.load('DICEGalaxyDisk/output_00001/info_00001.txt')
pp1 = yt.ParticlePlot(ds, ('all', 'particle_position_x'), ('all', 'particle_position_z'), ('all', 'particle_mass'))
pp1.save()
pp2 = yt.ParticlePlot(ds, ('all', 'particle_position_z'), ('all', 'particle_position_x'), ('all', 'particle_mass'))
pp2.save()
```
**Actual outcome**

pp1 gives `info_00001_2d-Profile_particle_position_x_particle_position_z_particle_mass.png`
![info_00001_2d-profile_particle_position_x_particle_position_z_particle_mass](https://user-images.githubusercontent.com/5411875/35686268-2f12b8bc-076c-11e8-8fa2-8cc3c8be8ccf.png)
pp2 gives `info_00001_Particle_y_particle_mass.png`
![info_00001_particle_y_particle_mass](https://user-images.githubusercontent.com/5411875/35686269-2f339fb4-076c-11e8-8f27-1b858407e7c2.png)

**Expected outcome**

I would expect to get two `ParticleProjectionPlot` with the axes swapped.

**Some information**
The bug is caused by https://github.com/yt-project/yt/blob/609c600ee28764dbb6d3fd29dd2f5a11943f585a/yt/visualization/particle_plots.py#L501-L528
This piece of code only recognises plots when the axis are in the right order. This should be replaced by something more flexible that accepts either (`y`, `x`) or (`x`, `y`).

**Comment by git-abhishek** [comment]
Hi @ngoldbaum @cphyc,
I am a GSoc'18 aspirant and would like to contribute to yt.
In order to understand yt, I tried to reproduce and fix this issue and I was able to get ParticleProjectionPlot for both the plots as shown in the screenshot
![capture](https://user-images.githubusercontent.com/36498066/36297957-7fc8ca5a-12c3-11e8-8912-4ff47bc1f7b4.PNG)

Let me know, if I can raise a pull request for the same.
Thank you!
